### ISSUE TITLE
Why is it impossible to deploy a supervisor in Ubuntu

### ISSUE THREAD

**Issue by Jinxiansen** [vapor/vapor#1383]
🚀
                                                                      Bug [REMOVE]
------------------------------------------------------------------------------------------------------

* Vapor Toolbox: 3.1.2
* Vapor Framework: 2.2.2
* OS: Ubuntu 16.04

<img width="576" alt="log" src="https://user-images.githubusercontent.com/16829428/34300130-a7ce2924-e761-11e7-928c-e4fa41d99121.png">
<img width="689" alt="nginx" src="https://user-images.githubusercontent.com/16829428/34300131-a80f82d4-e761-11e7-94d7-88a645b0d13b.png">
<img width="682" alt="run" src="https://user-images.githubusercontent.com/16829428/34300132-a8485f00-e761-11e7-994c-435285750b75.png">
<img width="676" alt="supervisor" src="https://user-images.githubusercontent.com/16829428/34300134-a88124de-e761-11e7-841e-b07aebc893f3.png">

These are the deployment information as well as the output of the error information, when I try to use **sudo supervisorctl restart all** the time, always fail.
Please give advice or comments。
```
English is not my native language; please excuse typing errors. 💪
```

**Comment by joscdk** [comment]
Can you share your supervisor config file?

**Comment by Jinxiansen** [comment]
@Joannis  ok ,Just a moment .

**Comment by joscdk** [comment]
But could look like the user you try to run your Vapor app through supervisor don't have access to the files

**Comment by Jinxiansen** [comment]
<img width="604" alt="config" src="https://user-images.githubusercontent.com/16829428/34300306-606c1270-e762-11e7-9a24-39bafe218345.png">

**Comment by joscdk** [comment]
Yep you are trying to run the vapor app as www-data, but you are using ubuntu user.

Either give www-data access to the files, or run the vapor app as ubuntu :) 

That should fix it :)

**Comment by Jinxiansen** [comment]
Yes, I want to run the vapor app as ubuntu, but the supervisor cannot deploy successfully. 😞

**Comment by joscdk** [comment]
Have you tried changing L4 in your supervisor file to

```
user=ubuntu
```

**Comment by Jinxiansen** [comment]
When I modify, re-execute the supervisor's reread, reload, or the same error message.

**Comment by Jinxiansen** [comment]
<img width="676" alt="2" src="https://user-images.githubusercontent.com/16829428/34300589-ba845c3a-e763-11e7-8ebe-33025d92ee66.png">

**Comment by Jinxiansen** [comment]
@joscdk  
<img width="703" alt="8080" src="https://user-images.githubusercontent.com/16829428/34301306-028b9c5c-e767-11e7-86b1-29468f5679fe.png">

<img width="927" alt="start" src="https://user-images.githubusercontent.com/16829428/34301318-0d5987de-e767-11e7-96d5-714587b287a1.png">


When I look at the helloWorldtdout.log log file and found that port 8080 is already up and running at the moment, I am extremely angry, I immediately find the 8080 PID, and the end of the process.
As shown, when I used ** sudo supervisorctl restart all ** again, I finally saw that the process started successfully.
thank you very much for your help! Solved this puzzled world puzzle that made me confused for two days!🤠🤠🤠
